Add enable required LLE modules for online features option and other improvements

PabloMK7 opened the pull request:
- Added option to enable required LLE modules for online features.
- (Android) Fixed bug that would cause FS RenameFile to fail sometimes on Android.
- (Android) Moved New 3DS mode and LLE applets to system settings tab on Android.
- (Android ) Fixed cfg save data related issues (mostly Console ID).
- (Android) Added MAC address setting from QT.
- ~~(Android) Fixed radio buttons not having a default selection on the Azahar setup prompt.~~
- Made AM title scanning asynchronous, which makes game boot way faster on Android on most cases.
- Made more AM functions asynchronous, to prevent stutter.
- Fixed bug in SOC that could cause the emulator to crash when disconnecting.
- Fixed keys not being initialized when processing console unique files.

DaniElectra commented on src/common/hacks/hack_list.cpp:
```diff
@@ -70,5 +70,43 @@ HackManager hack_manager = {
+                     0x000400100002C000, // USA
+                     0x000400100002C100, // EUR
+
+                     // System Settings
```
Could this conflict with the initial setup procedure?

PabloMK7 replied:
Nope, that is handled separately.

PabloMK7 committed:
Add LLE modules for online features option + AM Refactor w/ various improvements/fixes

- Added option to enable required LLE modules for online features.
- (Android) Fixed bug that would cause FS RenameFile to fail sometimes on Android.
- (Android) Moved New 3DS mode and LLE applets to system settings tab on Android.
- (Android) Fixed cfg save data related issues (mostly Console ID).
- Made AM title scanning asynchronous, which makes game boot way faster on Android on most cases.
- Made more AM functions asynchronous, to prevent stutter.
- Fixed bug in SOC that could cause the emulator to crash when disconnecting.
- Fixed keys not being initialized when processing console unique files.

PabloMK7 committed:
android: Added MAC address setting from QT

OpenSauce04 commented on src/android/app/src/main/java/org/citra/citra_emu/NativeLibrary.kt:
```diff
@@ -567,7 +567,11 @@ object NativeLibrary {
     fun createDir(directory: String, directoryName: String): Boolean =
         if (FileUtil.isNativePath(directory)) {
-            CitraApplication.documentsTree.createDir(directory, directoryName)
+            try {
+                CitraApplication.documentsTree.createDir(directory, directoryName)
+            } catch (e: Exception) {
+                false
+            }
```
Shouldn't this be logged?

OpenSauce04 commented on src/android/app/src/main/java/org/citra/citra_emu/NativeLibrary.kt:
```diff
@@ -641,7 +645,11 @@ object NativeLibrary {
     fun renameFile(path: String, destinationFilename: String): Boolean =
         if (FileUtil.isNativePath(path)) {
-            CitraApplication.documentsTree.renameFile(path, destinationFilename)
+            try {
+                CitraApplication.documentsTree.renameFile(path, destinationFilename)
+            } catch (e: Exception) {
+                false
+            }
```
Same as above.

OpenSauce04 committed:
configure_system.ui: Fixed indentation
